src/charm.py

```diff
@@ -343,6 +345,8 @@ def _store_certificates_from_config(self) -> None:
                 base64.b64decode(certificate_config).decode("utf-8").strip(),
             ]
         self._store_config_ca_chain(ca_chain_list)
+        self.tls_certificates.revoke_all_certificates()
```

Comment:
I am not convinced `revoke_all_certificates` should be inside of `_store_certificates_from_config`. The scope of the method was to store user provided certs inside of peer relation data. I thing `revoke_all_certificates` can be called directly from the `_config_changed` method

Reply:
Good point, moved.